lib/WorseReflection/Core/Inference/Walker/BinaryExpressionWalker.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Phpactor\WorseReflection\Core\Inference\Walker;
+
+use Microsoft\PhpParser\Node;
+use Microsoft\PhpParser\Node\Expression\BinaryExpression;
+use Phpactor\WorseReflection\Core\Inference\Frame;
+use Microsoft\PhpParser\Node\Expression\Variable;
+use Phpactor\WorseReflection\Core\Inference\NodeContext;
+use Phpactor\WorseReflection\Core\Inference\FrameResolver;
+use Phpactor\WorseReflection\Core\Inference\Symbol;
+use Phpactor\WorseReflection\Core\Inference\NodeContextFactory;
+use Phpactor\WorseReflection\Core\Inference\Variable as WorseVariable;
+use Microsoft\PhpParser\Node\Statement\ExpressionStatement;
+use Phpactor\WorseReflection\Core\Util\NodeUtil;
+
+class BinaryExpressionWalker extends AbstractWalker
+{
+    public function nodeFqns(): array
+    {
+        return [BinaryExpression::class];
+    }
+
+    public function walk(FrameResolver $resolver, Frame $frame, Node $node): Frame
```

Comment:
the existence of this class is necessary but puts into question the entire concept of the `Walker` classes.